### PR TITLE
feat(cb2-7727): handle vehicle type change for small trailers

### PR DESCRIPTION
**Pull request by tahayazici-bjss**
## Manage small trailers in VTM

- Display "SMALL TRL" in the tech record title when viewing a TRL with O1 EU vehicle category.
- Display SMALL TRL sections in the tech record summary when viewing a TRL with O1 EU vehicle category.
- Restrict selecting the O1 EU vehicle category for TRLs.
- When choosing to change vehicle type to SMALL TRL, set the type to TRL and the EU vehicle category to O1.

[CB2-7727](https://dvsa.atlassian.net/browse/CB2-7727)

### CODE REVIEW

File: src/app/features/tech-record/components/tech-record-amend-vrm/tech-record-amend-vrm.component.ts [hunk 1]
```diff
@@ -63,12 +65,10 @@ export class AmendVrmComponent implements OnInit {
...
-  ngOnInit(): void {
-    if (!this.techRecord) {
-      this.navigateBack();
-    }
-
-    this.actions$.pipe(ofType(updateTechRecordsSuccess), take(1)).subscribe(() => this.navigateBack());
+  get vehicleType(): VehicleTypes | undefined {
+    return this.techRecord?.vehicleType === VehicleTypes.TRL && this.techRecord.euVehicleCategory === EuVehicleCategories.O1
+      ? VehicleTypes.SMALL_TRL
+      : this.techRecord?.vehicleType;
```
Comment (gjulien-bjss) [inline src/app/features/tech-record/components/tech-record-amend-vrm/tech-record-amend-vrm.component.ts:71]: I would suggest putting this function to get the vehicle type in the tech record service. Ideally we would be removing this "dodgy" behaviour at some point, and this separation of concerns/abstraction would probably help that process. Plus it means less repeated code seeing as I've seen this in a few components already.

File: src/app/forms/templates/small-trailer/small-trailer-tech-record.template.ts [hunk 2]
```diff
@@ -65,12 +65,10 @@ export const SmallTrailerTechRecord: FormNode = {
...
       name: 'euVehicleCategory',
       label: 'EU vehicle category',
-      value: '',
+      value: EuVehicleCategories.O1,
       type: FormNodeTypes.CONTROL,
-      editType: FormNodeEditTypes.SELECT,
-      width: FormNodeWidth.S,
-      options: getOptionsFromEnum(EuVehicleCategories),
-      validators: []
+      editType: FormNodeEditTypes.HIDDEN,
```
Comment (gjulien-bjss) [inline src/app/forms/templates/small-trailer/small-trailer-tech-record.template.ts:70]: Maybe disabling it would be better than hiding it? Not sure

Comment (tahayazici) [reply src/app/forms/templates/small-trailer/small-trailer-tech-record.template.ts:70]: Suggestion applied.

File: src/app/features/tech-record/components/tech-record-amend-vrm/tech-record-amend-vrm.component.ts (hunk 1 above)
Comment (tahayazici) [reply src/app/features/tech-record/components/tech-record-amend-vrm/tech-record-amend-vrm.component.ts:71]: Suggestion applied.

**Review by shivangidas — APPROVED** [review]

**Review by gjulien-bjss — APPROVED** [review]

**Pull request merged**